libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -90,6 +90,46 @@ void StaticAnalyzer::endVisit(FunctionDefinition const&)
 	m_localVarUseCount.clear();
 }
 
+bool StaticAnalyzer::visit(ModifierInvocation const& _modifier)
+{
+	if (!m_constructor)
+		return true;
+
+	bool const v050 = m_currentContract->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050);
+
+	if (auto contract = dynamic_cast<ContractDefinition const*>(_modifier.name()->annotation().referencedDeclaration))
+		for (auto const &currentContract: m_currentContract->annotation().linearizedBaseContracts)
+			for (auto const& specifier: currentContract->baseContracts())
+			{
+				auto parent = specifier->name().annotation().referencedDeclaration;
+				if (contract == parent)
+				{
+					if (!specifier->arguments().empty())
+					{
+						SecondarySourceLocation ssl;
+						ssl.append("Overriden constructor call is here:", specifier->location());
+
+						if (v050)
+							m_errorReporter.declarationError(
+								_modifier.location(),
+								ssl,
+								"Duplicated super constructor call."
+							);
+						else
+							m_errorReporter.warning(
+								_modifier.location(),
+								"Duplicated super constructor calls are deprecated.",
+								ssl
+							);
+					}
+					else
+						return true;
```

Comment:
Is it really ok to stop searching the inheritance hierarchy here? Please add a test for:
```
contract C { constructor(uint) {} }
contract A is C(2) {}
contract B is C(2) {}
contract D is A, B { constructor() C(3) }
```

Perhaps some more tests including multiple inheritance would be in order :)

Reply:
For example, is the case
```
contract C { constructor(uint) {} }
contract A is C(2) {}
contract B is C(2) {}
```
covered?